library/spdm_responder_lib/libspdm_rsp_measurements.c

```diff
@@ -423,8 +408,10 @@ libspdm_return_t libspdm_get_response_measurements(void *context,
             }
         }
     } else {
+        fill_response_ptr = (void *)((uint8_t *)spdm_response +
+                                     spdm_response_size - measurements_no_sig_size);
         if(!libspdm_create_measurement_opaque(spdm_context, spdm_response,
-                                              spdm_response_size)) {
+                                              spdm_response_size, fill_response_ptr)) {
```

Comment:
inserting nonce and opaque fields in MEASURENTS is mandatory. 
So, it is better to move to out of  if statement and remove "libspdm_create_measurement_opaque()" from "libspdm_create_measurement_signature()."

Reply:
Yes, inserting nonce and opaque fields in MEASURENTS is mandatory.  But code should ensure create opaque and nonce successfully. So add the check in if statement.

About why add `libspdm_create_measurement_opaque()` in `libspdm_create_measurement_signature()`,it can reduce the redundant code.  Thanks.